### ISSUE TITLE
Minimize to tray functionality

### ISSUE THREAD

**Issue by cenobite1** [cboxdoerfer/fsearch#52]
Would it be possible to implement functionality that when one presses "ESC" the app minimizes to tray?

Thanks.

**Comment by cboxdoerfer** [comment]
Yes, that's possible and already planned. I just need to figure out how to implement it in order to support most or all desktops (plasma5, unity, xfce, ...).

**Comment by gersonjferreira** [comment]
Additionally, I would like it to minimize to tray automatically after I double-click on a file.

**Comment by l0rraine** [comment]
Has it come to true?

**Comment by cboxdoerfer** [comment]
@l0rraine, no not yet. With the GNOME developers removing the tray in GNOME Shell and the deprecation of the status icon API in GTK+, at least I don't have to bother with their crap anymore. So it seems we either need to go for Appindicator or StatusNotifier but I don't know how well they are supported on the common platforms. Need to do some research on that.

**Comment by gregorylearns** [comment]
+1 would like to request this functionality as well

**Comment by andrehtgti2** [comment]
+1 would like to request this functionality as well

**Comment by MostHated** [comment]
It is nice that it will minimize to the active application area of the bar, any chance there could be an always running daemon/ taskbar icon with a global hotkey that pulls up the search window similar to https://github.com/stnkl/EverythingToolbar ?

It would be great if it ran like a daemon, to always keep the search results refreshed in the background, similar to Albert. Have it do a full search refresh upon first opening it, unfortunately, feels like it... not defeats the purpose, but is more like a step in the wrong direction toward the goal of being able to quickly find what you are after. If it ran as a taskbar daemon upon login and then just refreshed the search index now and then, if you say, pressed ctrl + alt + f at any time, it could pop up, fully loaded, indexed, and ready to go.

Thanks,
-MH

**Comment by nekohayo** [comment]
Here's some updated information for implementation reference.

[Since 2023](https://discourse.flathub.org/t/portals-1-16-0-release-is-out/3404), the way forward to implement this in a Wayland-compatible and sandboxing-compatible (i.e. Flatpak) way is to use the FreeDesktop [background apps portal](https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.Background.html) standard (`org.freedesktop.portal.Background`).

This would allow the app to better let the operating system (and user) know when it's running in the background, and desktops to potentially better integrate it, now and in the future.

Historical context:
* https://blogs.gnome.org/aday/2017/08/31/status-icons-and-gnome/
* https://blog.tingping.se/2019/09/07/how-to-design-a-modern-status-icon.html
* Future UX in GNOME: https://gitlab.gnome.org/Teams/Design/os-mockups/-/work_items/191

This FreeDesktop specification is already implemented by a couple of desktop environments (if [this wiki page](https://wiki.archlinux.org/title/XDG_Desktop_Portal) is any indication, this is at least supported by KDE, Deepin, Cinnamon, GNOME, elementaryOS/Pantheon, and possibly others).

My general impression is that this would be the more future-proof, sustainable and natively-integrated way to handle the traditional "minimize to system tray icon" usecase in the long term, particularly for applications running inside Flatpak. It lets the app officially keep running in the background without a visible window, _without_ its process getting killed by the Flatpak sandbox for security reasons.

It also lets you do things like specifying the reason why it is running, using the `SetStatus` function so the desktop environment UI can also expose this for the user. For example, Pika Backup says "Monitoring Backup Schedule", Shortwave says "Playing [the radio station name]", etc.

---

Then, if you also want the app to be activatable from a global keyboard shortcut while it's in the background, you also need to use the [Global Shortcuts](https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.GlobalShortcuts.html) portal.

---

I would suggest retitling this to prefix (or suffix) the existing title to mention implementing those two portals.
